crates/net/src/sim2h_worker.rs

```diff
@@ -374,23 +382,32 @@ impl NetWorker for Sim2hWorker {
                 if self.time_of_last_connection_attempt.elapsed() > RECONNECT_INTERVAL {
                     self.time_of_last_connection_attempt = Instant::now();
                     warn!("No connection to sim2h server. Trying to reconnect...");
-                    self.stream_manager
-                        .connect(&self.server_url.clone().into())?;
+                    if let Err(e) = self.stream_manager.connect(&self.server_url.clone().into()) {
+                        error!("TransportError trying to connect to sim2h server: {:?}", e);
+                    }
                 }
             }
             ConnectionStatus::Initializing => debug!("connecting..."),
             ConnectionStatus::Ready => {
-                let client_messages = self.inbox.drain(..).collect::<Vec<_>>();
-                for data in client_messages {
-                    debug!("CORE >> Sim2h: {:?}", data);
-                    if let Err(error) = self.handle_client_message(data) {
-                        error!("Error handling client message in Sim2hWorker: {:?}", error);
-                    }
-                    did_something = true;
+                let previously_failed_messages =
+                    self.outgoing_failed_messages.drain(..).collect::<Vec<_>>();
+                for message in previously_failed_messages {
+                    // Ignore error here since send_wire_message logs TransportErrors during send
+                    // and puts back the failed message into self.outgoing_failed_messages
+                    let _ = self.send_wire_message(message);
                 }
             }
         }
 
+        let client_messages = self.inbox.drain(..).collect::<Vec<_>>();
+        for data in client_messages {
+            debug!("CORE >> Sim2h: {:?}", data);
```

Comment:
took me a minute to figure out what was going on here, perhaps a comment like:

```suggestion
            debug!("CORE >> Sim2h: {:?}", data);
            // outgoing messages triggered by `self.hand_client_message` that fail because of
            // connection status, will automatically be re-sent via `self.outgoing_failed_messages`
```